samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -583,27 +591,38 @@ static void irq_handler(struct device *gpiob, struct gpio_callback *cb,
 	/* Enter active polling mode until no motion */
 	gpio_pin_disable_callback(gpio_dev, OPTICAL_PIN_MOTION);
 
-	if (atomic_cas(&state, STATE_IDLE, STATE_FETCHING)) {
+	switch (state.state) {
+	case STATE_IDLE:
 		/* Wake up thread */
+		state.state = STATE_FETCHING;
+		state.sample = true;
 		k_sem_give(&sem);
-	} else if (atomic_get(&state) == STATE_SUSPENDED) {
+		break;
+
+	case STATE_SUSPENDED:
+		LOG_ERR("Motion wake up");
```

Comment:
Should it be LOG_ERR here ?

Reply:
well spotted - this is a debug message I forgot to remove...

Reply:
Fixed